src/lib/Libutil/misc_utils.c

```diff
@@ -605,6 +605,69 @@ pbs_asprintf(char **dest, const char *fmt, ...)
 	return len;
 }
 
+/**
+ * @brief
+ *	concatenates formatted string to (dest) of size (size) at index (idx),
+ *	reallocating dest if needed.
+ *
+ * @param[in, out] dest - character pointer that will point to allocated
+ *			  space ** must be freed by caller **
+ * @param[in, out] size - size of dest. Will be modified if reallocated
+ * @param[in, out] idx - index at the end of the string.
+ * @param[in] fmt - format for printed string
+ * @param[in] ... - arguments to format string
+ *
+ * @return int
+ * @retval -1 - Error
+ * @retval >=0 - Length of new string, not including terminator
+ */
+int
+pbs_asprintcatf(char **dest, size_t *size, size_t *idx, const char *fmt, ...)
+{
+	va_list args;
+	int len;
+	size_t new_size = *size;
+	char * newbuf = NULL;
+
+	if (!dest)
+		return -1;
+	if (!fmt)
+		return -1;
+	va_start(args, fmt);
+
+	if (*dest == NULL)
+		*dest = malloc(new_size);
+	if (*dest == NULL) {
+		len = -1;
+		goto pbs_asprintcatf_exit;
+	}
+
+	len = pbs_asprintf_len(fmt, args);
+	if (len < 0)
+		goto pbs_asprintcatf_exit;
+
+	while (*idx + len > new_size) {
+		new_size *= 2;
+	}
+	if (new_size != *size) {
+		newbuf = realloc(*dest, new_size);
+		if (newbuf == NULL) {
+			len = -1;
```

Comment:
So if we allocate dest in this function and then fails to realloc it then do we expect the caller to free dest?

Reply:
That is correct. I'll add it to the docstring